test/src/pages/details/desktop-view/instructor-resource-tab.test.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import {render, screen} from '@testing-library/preact';
+import InstructorResourceTab from '~/pages/details/desktop-view/instructor-resource-tab/instructor-resource-tab';
+import * as DH from '~/helpers/use-document-head';
+import bookData from '../../../data/college-algebra-postproc';
+import mockRawResourceData from '../../../data/raw-resources';
+import {DetailsContextProvider} from '~/pages/details/context';
+import ShellContextProvider from '../../../../helpers/shell-context';
+
+const mockUseUserContext = jest.fn();
+const mockNavigate = jest.fn();
+const mockDataFromSlug = jest.fn(() => mockRawResourceData);
+
+jest.mock('~/contexts/user', () => ({
+    __esModule: true,
+    default: () => mockUseUserContext(),
+    UserContextProvider: ({children}: any) => children // eslint-disable-line
+}));
+jest.spyOn(DH, 'setPageTitleAndDescriptionFromBookData').mockReturnValue();
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    __esModule: true,
+    useNavigate: () => ({navigate: () => mockNavigate()})
+}));
+
+jest.mock('~/helpers/page-data-utils', () => ({
+    ...jest.requireActual('~/helpers/page-data-utils'),
+    __esModule: true,
+    useDataFromSlug: () => mockDataFromSlug()
+}));
+
+jest.mock('~/pages/details/common/resource-box/left-content');
+
+function Component({data=bookData}) {
+    return (
+        <ShellContextProvider>
+            <DetailsContextProvider contextValueParameters={{data}}>
+                <InstructorResourceTab />
+            </DetailsContextProvider>
+        </ShellContextProvider>
+    );
+}
+
+describe('details/instructor-resource-tab', () => {
+    beforeEach(() => {
+        mockUseUserContext.mockReturnValue({
+            userStatus: {
+                isInstructor: false
+            }
+        });
+    });
+    it('returns null until user status exists', async () => {
+        mockUseUserContext.mockReturnValue({});
+        render(<Component />);
+        let caught = false;
+
+        try {
+            await screen.findByRole('heading');
```

Comment:
Maybe you can use`queryByRole` on these, it returns null for a miss. I'm always worried about try/catch swallowing other errors.

Reply:
I use `find` because there's some promise-resolving that happens. For example, in the next test, I expect a `heading`, but have to wait for it. Here, I wait to be sure it's not showing up.

Reply:
Oh interesting, maybe then `await expect(screen.findByRole('heading')).rejects.toThrow()` would work, it kinda feels more idiomatic to jest?

Reply:
Much nicer, yes. I've replaced the try/catch construct throughout my tests now.